README.md

```diff
@@ -93,6 +95,36 @@ Then you should be able to run:
 
 And browse the MFE at http://apps.local.overhang.io:3001/home/
 
+</details>
+
+## Running an MFE in development mode
+
+To run an MFE in development mode, you would need to clone it and to mount it's directory
+
+ Assuming you need to modify `frontend-app-account`.
+
+1. Clone it _if it's not already_ 
+2. `npm install` _make sure you are on the correct node version `node --version` it shall match `cat .nvmrc`_
+3. `tutor mounts add "./mfe/frontend-app-account"`. 
+
+### How do I override specific npm pks?
+
+The following is an example of overriding a header
+
+1. Clone it _if it's not already_
+2. `npm install` _make sure you are on the correct node version `node --version` it shall match `cat .nvmrc`_
+3. Mount the pkg to the container `tutor  mounts add "account:./mfe/frontend-component-header:/openedx/frontend-component-header"` scheme: `service:host_path:containter_path`
+3. edit the `pacakage.json` file of the MFE you want to override its dependency, for example assuming you apply your local version of header for the account MFE for previous step:
```

Comment:
I would be really nice if we could avoid MFE source to run in dev mode

Reply:
I was trying to mimic the workflow https://github.com/Abstract-Tech/community-theme-tutor/blob/main/bin/mfe_dev, should  I make an expectation about where typically dev clones the MFEs and their assests, may be same root directory  as tutor root?

Reply:
`bin/mfe_dev` workflow supposes that MFE (and header/footer/brand) source is located in `mfe/` directory. That's exatcly the reason why we have those as submodules in this repo. Let's keep it like that if possible.

Reply:
may be I don't get it, so isn't `mfe/` same as ** MFE source**?

Reply:
- `mfe/frontend-app-{{ MFE_NAME }}` for individual MFE source
- `mfe/frontend-component-header` for header
- `mfe/frontend-component-footer` for footer
- `mfe/openedx-brand` for brand

The later three are shared between all the MFEs

Reply:
Aha exactly that what I thought... may be you meant it would be better to not necessary change the `pacakge.json` of a particular MFE?

Reply:
Exactly. In `mfe_dev` workflow there is `module.config.js` to achieve that. I wonder we can use it with `tutor mounts ...`

Reply:
I will give it another shot

Reply:
I think it worked, instead of mutating the `pacakge.json` file mounting the depencey worked, though I got a webpack warning...

The steps would go like this

```bash
tutor mounts add ./mfe/frontend-app-account  tutor  mounts add account:./mfe/frontend-component-header:/openedx/frontend-component-header
tutor mounts add account:./mfe/frontend-component-header:/openedx/frontend-component-header
tutor  mounts add account:./mfe/frontend-component-footer:/openedx/frontend-component-footer 
tutor  mounts add account:./mfe/brand-openedx:/openedx/brand-openedx
tutor mounts add account:./mfe/module.config.js:/openedx/frontend-app-account/module.config.js
```

An each pkg, dev would need to `install and build` the depencey... but there is a ceveta with the example above, that is the path suppose to be absoulte, otherwise, when chaning directory tutor would break... I found it a bit confusing since tutor convert the MFE path to absoulte, _implicit versioun_ however for the latter this is not the case

Reply:
I ran following commands:

```
tutor mounts add ./mfe/frontend-app-account
tutor mounts add account:./mfe/frontend-component-header:/openedx/frontend-component-header
tutor mounts add account:./mfe/frontend-component-footer:/openedx/frontend-component-footer
tutor mounts add account:./mfe/brand-openedx:/openedx/brand-openedx
tutor mounts add account:./mfe/module.config.js:/openedx/frontend-app-account/module.config.js
```
Which resulted in this block being added to config.yml:

```
MOUNTS:
- /home/snglth/code/community-theme/community-theme-tutor/mfe/frontend-app-account
- account:./mfe/frontend-component-header:/openedx/frontend-component-header
- account:./mfe/frontend-component-footer:/openedx/frontend-component-footer
- account:./mfe/brand-openedx:/openedx/brand-openedx
- account:./mfe/module.config.js:/openedx/frontend-app-account/module.config.js
```
But that makes account to have some troubles finding header:
![image](https://github.com/Abstract-Tech/community-theme-tutor/assets/8686360/a20fae0b-a6f3-4929-8ab5-a53426df65e3)

I think that modules.config.js has no effect for some reason.

Reply:
Did you run `npm install` `npm run build` on the host in the pkgs... also need to do that for account mfe

Reply:
I have the same error before, but after `npm install` and `npm run build`... also if none works try to run `npm start` insdie the header in the host

Reply:
module.config.js was not applied because it was mounted to incorrect (for `tutor dev`) directory. Correct mount line would be:

```
- account:./mfe/module.config.js:/openedx/app/module.config.js
```

Even after that have been getting issues with dependencies:
![image](https://github.com/Abstract-Tech/community-theme-tutor/assets/8686360/fd16ffd6-9ae5-4225-b14a-e708798d63da)

But it was fixed by restarting account container. Looks very promising now! 

The missing pice now is to automate all the preparation steps with `npm install` && `npm run build` and so on just like `mfe_dev` [does](https://github.com/Abstract-Tech/community-theme-tutor/blob/main/bin/mfe_dev#L25)
 
Also it looks like we can have all of our MOUNTS relative. At least this works for me:

```
MOUNTS:
- ./mfe/frontend-app-account
- account:./mfe/frontend-component-header:/openedx/frontend-component-header
- account:./mfe/frontend-component-footer:/openedx/frontend-component-footer
- account:./mfe/brand-openedx:/openedx/brand-openedx
- account:./mfe/module.config.js:/openedx/app/module.config.js
```

Reply:
> Also it looks like we can have all of our MOUNTS relative. At least this works for me:

 Yes you can but if you change your directory and then hit `tutor config save` and restart, it will erroed, because tutor/docker will try to mount relative to your current directory which might change

Reply:
It doesn't fail for me. In `env/dev/docker-compose.yml` tutor makes all the paths absolute:

```
...
  account:
      image: "docker.io/overhangio/openedx-account-dev:16.0.0"
      ports:
          - "1997:1997"
      stdin_open: true
      tty: true
      volumes:
          - ../plugins/mfe/apps/mfe/webpack.dev-tutor.config.js:/openedx/app/webpack.dev-tutor.config.js:ro
          - /home/snglth/code/community-theme/community-theme-tutor/mfe/frontend-app-account:/openedx/app
          - /home/snglth/code/community-theme/community-theme-tutor/mfe/frontend-component-header:/openedx/frontend-component-header
          - /home/snglth/code/community-theme/community-theme-tutor/mfe/frontend-component-footer:/openedx/frontend-component-footer
          - /home/snglth/code/community-theme/community-theme-tutor/mfe/brand-openedx:/openedx/brand-openedx
          - /home/snglth/code/community-theme/community-theme-tutor/mfe/module.config.js:/openedx/app/module.config.js
      restart: unless-stopped
      depends_on:
          - lms
...
```

Reply:
Humm.. It looks correct (_The rendering_),  I am a bit confuse, the thing when I reviewed tutor code, it seems that it would only convert it absoulte when it's implicit, hence: https://github.com/overhangio/tutor/blob/8de5edfab735d077fab7b2694d668354caeeead7/tutor/commands/mounts.py#L104-L108 

What is the output of `tutor mounts list`

Also it could be a problem just with Mac/Zsh

Reply:
I also use zsh on linux. `tutor mounts list` produces:

```
-> tutor mounts list
- name: ./mfe/frontend-app-account
  build_mounts:
  - image: mfe
    context: account-src
  - image: account-dev
    context: account-src
  compose_mounts:
  - service: account
    container_path: /openedx/app
- name: account:./mfe/frontend-component-header:/openedx/frontend-component-header
  build_mounts: []
  compose_mounts:
  - service: account
    container_path: /openedx/frontend-component-header
- name: account:./mfe/frontend-component-footer:/openedx/frontend-component-footer
  build_mounts: []
  compose_mounts:
  - service: account
    container_path: /openedx/frontend-component-footer
- name: account:./mfe/brand-openedx:/openedx/brand-openedx
  build_mounts: []
  compose_mounts:
  - service: account
    container_path: /openedx/brand-openedx
- name: account:./mfe/module.config.js:/openedx/app/module.config.js
  build_mounts: []
  compose_mounts:
  - service: account
    container_path: /openedx/app/module.config.js
```

Reply:
yup that what I expected, it would save it as relative but then if you did `tutor config save`  _while in other directory_ or if you initially added in a directory other than tutor root, would it still work? 

In any case this might not be a very serious issue for us, if we are going to be always conssitent where the componenets are realtive to tutor root

Reply:
I have updated the gudie to use `module.config.js` instead of pacakge.json per last commit #[924cdb8](https://github.com/Abstract-Tech/community-theme-tutor/pull/41/commits/924cdb8cdef2d898df390848919661041e24a9cb)

Reply:
Looks good to me